### ISSUE TITLE
Convolution3DReparameterization logits exploding

### ISSUE THREAD

**Issue by mrivar** [tensorflow/probability#288]
I am using 3D Convolutional Reparametrization networks to solve an Action Recognition problem ([model here](https://github.com/mrivar/infrared-AR/blob/master/model/_3DBCNN_infar.py)).
However, the value of my logits seems to be exploding. I get really high values, making my loss way too big as well (>= 1 million, only cross entropy therefore ignoring KL). In addition to this, the network does not seem to be learning, as the accuracy doesn't really change.
Using the exact same [model with regular keras Conv3D layers](https://github.com/mrivar/infrared-AR/blob/master/model/_3DCNN_infar.py) seems to work perfectly fine, going up to 20% accuracy after just a few steps and getting around 50% after some epochs.

I am using the [InfAR dataset](https://www.sciencedirect.com/science/article/pii/S0925231216307044) and my code is public in the following repo: [mrivar/infrared-AR](https://github.com/mrivar/infrared-AR). [main file](https://github.com/mrivar/infrared-AR/blob/master/main.py).

An example on the logits results I get:
```
logits = array([[ -119181.83,   321628.  ,   527259.2 ,   694675.1 ,   -43457.19,
         -201538.7 ,   808418.  ,   328285.  ,  -574974.56, -1367197.9 ,
          221278.66,  -219717.78]], dtype=float32)

loss_value = 927599.8
```

What could be wrong? The loss calculation and optimization, and the code structure are the same as in the examples [bayesian_neural_network](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/bayesian_neural_network.py) and [cifar10_bnn](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/cifar10_bnn.py), just with 3D CNNs.

**Comment by mrivar** [comment]
The models started working (i.e. learning) when I sued more complex ones (s.a. [ResNet18](https://github.com/mrivar/infrared-AR/blob/master/model/resnet18_3d_bayesian.py)) and I set a higher learning rate (previously it was on 1e-5, now on 1e-4).
I guess the information was too complex for the model to learn?
